internal/metrics/queue.go

```diff
@@ -128,7 +129,8 @@ func (p PodHistograms) Inc(key PodTags) {
 
 // MarkTimes inspects pod conditions and marks lifecycle transition times
 func (p PodHistograms) MarkTimes(ctx context.Context, pod *v1.Pod, instanceGroupTagLabel string, now time.Time) {
-	instanceGroupTag := InstanceGroupTag(ctx, pod.Spec.NodeSelector[instanceGroupTagLabel])
+	ig, _ := internal.FindInstanceGroupFromPodSpec(pod.Spec, instanceGroupTagLabel)
```

Comment:
lets do what we are doing in `resource.go` and check err, and set to `""` explicitly

Reply:
done.